okhttp/src/main/java/okhttp3/OkHttpTrustManager.kt

```diff
@@ -0,0 +1,159 @@
+package okhttp3
+
+import android.annotation.SuppressLint
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.net.Socket
+import java.security.cert.Certificate
+import java.security.cert.X509Certificate
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.X509ExtendedTrustManager
+import javax.net.ssl.X509TrustManager
+import okhttp3.internal.platform.Platform
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
+
+interface OkHttpTrustManager : X509TrustManager {
+  fun checkServerTrusted(
+    chain: Array<out X509Certificate>,
+    authType: String,
+    host: String
+  ): List<Certificate>?
+
+  val delegate: X509TrustManager
```

Comment:
Maybe this should change to a Function `(String) -> X509TrustManager` and delete allowedPredicate.